services/ux-backend/handlers/onboarding_tokens.go

```diff
@@ -0,0 +1,120 @@
+package handler
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/red-hat-storage/ocs-operator/v4/services/types"
+	"k8s.io/klog/v2"
+)
+
+const onboardingPrivateKeyFilePath = "/etc/private-key/key"
+
+func OnboardingTokensHandler(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+
+	var err error
+	switch r.Method {
+	case "POST":
+
+		tokenExpirationDate := time.Now().
+			Add(time.Duration(tokenLifetimeInHours) * time.Hour).
+			Unix()
```

Comment:
This should be done inside the generate token function. IF not the generated token function who have no context about the given date will need to validate the date and return an error (if it is in the past, or less then 1 hour into the future). I always prefer to use delta times when providing information to nested code. This way the only thing that need validation is the sign of the value.

Reply:
moved to generate token function